iroha-cli/interactive/impl/interactive_query_cli.cpp

```diff
@@ -168,59 +173,67 @@ namespace iroha_cli {
           local_time_, creator_, counter_, account_id, asset_id);
     }
 
-    nonstd::optional<iroha::model::Pager> InteractiveQueryCli::parsePager(
+    iroha::model::Pager InteractiveQueryCli::parsePager(
         const std::string &encoded_tx_hash, const std::string &limit_str) {
-      return nonstd::make_optional<iroha::model::Pager>() |
-             [&encoded_tx_hash](iroha::model::Pager pager)
-               -> nonstd::optional<iroha::model::Pager> {
-               /**
-                * Use '-' as the argument of tx_hash to specify that
-                * the hash is empty because interactive cli splits
-                * arguments by space.
-                */
-               const auto PAGER_TX_HASH_EMPTY = "-";
-               if (encoded_tx_hash == PAGER_TX_HASH_EMPTY) {
-                 pager.tx_hash.fill(0);
-               } else {
-                 const auto decoded_hash =
-                   iroha::hexstringToBytestring(encoded_tx_hash);
-                 if (not decoded_hash) {
-                   return nonstd::nullopt;
-                 }
-                 pager.tx_hash = iroha::hash256_t::from_string(*decoded_hash);
-               }
-               return pager;
-             } | [&limit_str](auto pager) -> nonstd::optional<iroha::model::Pager> {
-        try {
-          pager.limit = boost::lexical_cast<decltype(pager.limit)>(limit_str);
-        } catch (...) {
-          return nonstd::nullopt;
+      iroha::model::Pager pager;
+      if (encoded_tx_hash.empty()) {
+        pager.tx_hash.fill(0);
+      } else {
+        const auto decoded_hash = iroha::hexstringToBytestring(encoded_tx_hash);
+        if (decoded_hash) {
+          pager.tx_hash = iroha::hash256_t::from_string(*decoded_hash);
+        } else {
+          log_->warn(
+              "Specified hash size is invalid. Set empty hash as "
+              "default");
+          pager.tx_hash.fill(0);
```

Comment:
Default hash looks bad, because of the following reasons:
* class shouldn't contain the logic of creation hash by itself
* default behavior in this case worse - not obvious usage
* if there was implied to make default strategy - this should be customizable - passes as a dependency.

Reply:
Yes, it is bad to change the value from client outside of the API implementation.
Maybe customizing hash256_t -> pager_hash_t and allows 0 and 32 byte size?

Reply:
There is no stateless validation in iroha-cli but converting hex to bytes is required to generate query because `iroha::hash256_t::from_string` throw exception if the size is not correct.

One solution seems that it is okay to use `optional` in the return value of `parsePager`. But because all queries return `std::shared_ptr` and `query_handlers_` referring such functions, it cannot propagate the error result when the conversion fails. (please see also: https://github.com/hyperledger/iroha/pull/830/files#diff-0810ba332c21013b016cd0dd5d525cdeL211)

So in CLI, I removed `nonstd::optional` and apply the default value of the hash and show warning log.